### ISSUE TITLE
Xcode Warnings

### ISSUE THREAD

**Issue by pwillsey** [AzureAD/azure-activedirectory-library-for-objc#898]
We are currently seeing about 50 warnings from the ADAL library, they all appear to be problems with method documentation, where the `@param` name doesn't match the method signature. Here's an example:

```objective-c
/*!
    Creates an instance of ADAuthenticationContext with the provided parameters.
 
    @param authority            The AAD or ADFS authority. Example: @"https://login.windows.net/contoso.com"
    @param validateAuthority    Specifies if the authority should be validated.
    @param error                (Optional) Any extra error details, if the method fails
 
    @return An instance of ADAuthenticationContext, nil if it fails.
 */
+ (ADAuthenticationContext*)authenticationContextWithAuthority:(NSString*)authority
                                             validateAuthority:(BOOL)validate
                                                         error:(ADAuthenticationError* __autoreleasing *)error;
```
In this example `@param validateAuthority` doesn't match the variable name `validate`. I'm happy to do a PR if necessary

**Comment by dtweston** [comment]
I just submitted #900 to fix these issues.

**Comment by partnerinflight** [comment]
Should be fixed in 2.4.0+
